test/Microsoft.AspNet.Localization.Tests/QueryStringRequestCultureProviderTest.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved. 
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Localization;
+using Microsoft.AspNet.TestHost;
+using Microsoft.Framework.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.Framework.Localization.Tests
+{
+    public class QueryStringRequestCultureProviderTest
+    {
+        [Fact]
+        public async void GetCultureInfoFromQueryString()
+        {
+            using (var server = TestServer.Create(app =>
+            {
+                var options = new RequestLocalizationOptions();
+                app.UseRequestLocalization(options);
+                app.Run(context =>
+                {
+                    var requestCultureFeature = context.GetFeature<IRequestCultureFeature>();
+                    var requestCulture = requestCultureFeature.RequestCulture;
+                    Assert.Equal("ar-SA", requestCulture.Culture.Name);
+                    Assert.Equal("ar-YE", requestCulture.UICulture.Name);
+                    return Task.FromResult(0);
+                });
+            },
+            services =>
+            {
+                services.AddLocalization();
+            }))
+            {
+                var client = server.CreateClient();
+                var response = await client.GetAsync("/page?culture=ar-SA&ui-culture=ar-YE");
+            }
+        }
+
+        [Fact]
+        public async void GetDefaultCultureInfoIfCultureKeysAreMissing()
+        {
+            using (var server = TestServer.Create(app =>
+            {
+                var options = new RequestLocalizationOptions();
+                app.UseRequestLocalization(options);
+                app.Run(context =>
+                {
+                    var requestCultureFeature = context.GetFeature<IRequestCultureFeature>();
+                    var requestCulture = requestCultureFeature.RequestCulture;
+                    Assert.Equal(options.DefaultRequestCulture.Culture.Name, requestCulture.Culture.Name);
+                    Assert.Equal(options.DefaultRequestCulture.UICulture.Name, requestCulture.UICulture.Name);
+                    return Task.FromResult(0);
+                });
+            },
+            services =>
+            {
+                services.AddLocalization();
+            }))
+            {
+                var client = server.CreateClient();
+                var response = await client.GetAsync("/page");
+            }
+        }
+
+        [Fact]
+        public async void GetDefaultCultureInfoIfCultureValuesAreInvalid()
```

Comment:
Can you make 2 tests out of this - one when UI Culture is invalid and other when Culture is invalid.